pymchelper/executor/runner.py

```diff
@@ -45,26 +51,52 @@ def __init__(self, jobs=None, keep_workspace_after_run=False, output_directory='
         self.workspace_manager = WorkspaceManager(output_directory=output_directory,
                                                   keep_workspace_after_run=keep_workspace_after_run)
 
-    def run(self, settings):
+    def run(self):
         """
         Execute parallel simulation processes, creating workspace (and working directories) in the `output_directory`
         In case of successful execution return True, otherwise return False
         """
         start_time = timeit.default_timer()
 
-        # SHIELD-HIT12A and Fluka require RNG seeds to be integers greater or equal to 1
-        # each of the workers needs to have its own different RNG seed
-        rng_seeds = range(1, self.jobs + 1)
+        if self.settings.simulator_type in [SimulatorType.shieldhit, SimulatorType.fluka]:
+            # SHIELD-HIT12A and Fluka require RNG seeds to be integers greater or equal to 1
+            # each of the workers needs to have its own different RNG seed
+            rng_seeds = range(1, self.jobs + 1)
+
+        elif self.settings.simulator_type == SimulatorType.topas:
+            # For TOPAS we don't need to create multiple working directories and a pool of workers,
+            # as we can use embedded parallelization in TOPAS.
+            # For that we need to modify the input file and set the number of threads to the number of jobs.
+            # We set one rng seed, so one working directory and one worker will be created.
+
+            modified_input_filename = Path(self.settings.input_path).name.replace(".txt", "_modified.txt")
+            modified_input_path = Path(self.settings.input_path).parent / modified_input_filename
+
+            with open(self.settings.input_path, 'r') as f:
+                config = f.read()
+                if "i:Ts/NumberOfThreads" in config:
+                    pattern = r"i:Ts/NumberOfThreads\s*=\s*\d+"
+                    replacement = f"i:Ts/NumberOfThreads = {self.jobs}"
+                    config = re.sub(pattern, replacement, config)
+                else:
+                    config = f"i:Ts/NumberOfThreads = {self.jobs}\n" + config
+
+            with open(modified_input_path, "w") as file:
+                file.write(config)
```

Comment:
```suggestion
            modified_input_path.write_text(config)
```

would that work as well ?

Reply:
Yes